src/slice.rs

```diff
@@ -0,0 +1,33 @@
+use crate::display_list::DisplayList;
+use std::fmt;
+
+#[derive(Debug, Clone, Default)]
+pub struct Slice<'s> {
+    pub source: &'s str,
+    pub line_start: Option<usize>,
+    pub origin: Option<&'s str>,
+    pub annotations: Vec<SourceAnnotation<'s>>,
```

Comment:
I guess this should be `&'s [SourceAnnotation<'s>]` since it's always user-supplied and not computed